src/nft/ZoraCreator1155Impl.sol

```diff
@@ -444,15 +446,48 @@ contract ZoraCreator1155Impl is
         // Require admin from the minter to mint
         _requireAdminOrRole(address(minter), tokenId, PERMISSION_BIT_MINTER);
 
+        // Get the token's first minter
+        address firstMinter = _handleFirstMinter(tokenId, minterArguments);
+
         // Get value sent and handle mint rewards
-        uint256 ethValueSent = _handleRewardsAndGetValueSent(msg.value, quantity, getCreatorRewardRecipient(), createReferrals[tokenId], mintReferral);
+        uint256 ethValueSent = _handleRewardsAndGetValueSent(
+            msg.value,
+            quantity,
+            getCreatorRewardRecipient(),
+            createReferrals[tokenId],
+            mintReferral,
+            firstMinter
+        );
 
         // Execute commands returned from minter
         _executeCommands(minter.requestMint(msg.sender, tokenId, quantity, ethValueSent, minterArguments).commands, ethValueSent, tokenId);
 
         emit Purchased(msg.sender, address(minter), tokenId, quantity, msg.value);
     }
 
+    /// @dev Get and/or set the first minter a token
+    function _handleFirstMinter(uint256 tokenId, bytes calldata data) internal returns (address) {
```

Comment:
very minor (and optional improvement) - for readability, having `returns (address firstMinter)` would be clearer.